File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/injection/modules/ConfigurationModule.java
package com.crpr.androidcinema.injection.modules;

import com.crpr.androidcinema.data.api.factories.ApiServiceFactory;
import com.crpr.androidcinema.data.api.services.ApiConfigurationService;
import com.crpr.androidcinema.domain.common.configuration.GetConfiguration;
import com.crpr.androidcinema.domain.common.configuration.GetConfigurationProcess;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by claudioribeiro on 09/07/16.
 */
@Module
public class ConfigurationModule {

    /************************* PROCESS *******************************/
    @Provides
    GetConfiguration.Process provideGetConfigurationProcess(GetConfiguration.Service service){
        return new GetConfigurationProcess(service);
    }

    /************************* SERVICE *******************************/
    @Provides
    @Singleton
    GetConfiguration.Service provideApiConfigurationService(ApiServiceFactory factory){
        return new ApiConfigurationService(factory.getServiceClient(ApiConfigurationService.ServiceClient.class));
    }

}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/injection/modules/ConfigurationModule.kt
package com.crpr.androidcinema.injection.modules

import com.crpr.androidcinema.data.api.factories.ApiServiceFactory
import com.crpr.androidcinema.data.api.services.ApiConfigurationService
import com.crpr.androidcinema.domain.common.configuration.GetConfiguration
import com.crpr.androidcinema.domain.common.configuration.GetConfigurationProcess
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by claudioribeiro on 09/07/16.
 */
@Module
class ConfigurationModule {

    /************************* PROCESS  */
    @Provides
    internal fun provideGetConfigurationProcess(service: GetConfiguration.Service): GetConfiguration.Process {
        return GetConfigurationProcess(service)
    }

    /************************* SERVICE  */
    @Provides
    @Singleton
    internal fun provideApiConfigurationService(factory: ApiServiceFactory): GetConfiguration.Service {
        return ApiConfigurationService(factory.getServiceClient(ApiConfigurationService.ServiceClient::class.java))
    }

}
